### PR TITLE
[3227] Add "Request PE courses for 2021/2022" section to provider page

### PR DESCRIPTION
**Pull request by AgaDufrat**
### Context
**User Story**
As an accredited body
I need to see the option to 'request PE courses for 2021/2022' when I log into Publish
So that I know where to go to request permission to recruit for PE in the next cycle

```
Given I am on the provider page
When I scroll to the bottom 
Then I see a link to "Request PE courses for 2020/21"
And content explaining what the feature is
When I click the "Request PE courses for 2020/21" text
Then I go to the "Request PE courses for 2020/21" page
```
### Changes proposed in this pull request
pe_allocations

See commits for details.

* The content appears at the bottom of the provider page under 'locations'
* The link takes me to the "Request PE courses for 2020/21"
* Available to admins only
* Page and link only exists for accredited bodies

### Guidance to review
- go to organisations page, click "Request PE courses for 2020/21"
- try for different users

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Product Review

### CODE REVIEW

**Comment by AgaDufrat** [comment]
[[S] 1. Add "Request PE courses for 2021/2022" section to provider page](https://trello.com/c/2iPQn0Tk/3227-s-1-add-request-pe-courses-for-2021-2022-section-to-provider-page)

File: app/helpers/breadcrumb_helper.rb [hunk 1]
```diff
@@ -69,4 +69,9 @@ def training_provider_courses_breadcrumb
     path = training_provider_courses_provider_recruitment_cycle_path(@provider.provider_code, @provider.recruitment_cycle_year, @training_provider.provider_code)
     training_providers_breadcrumb << ["#{@training_provider.provider_name}’s courses", path]
   end
+
+  def pe_allocations_breadcrumb
+    path = pe_allocations_provider_recruitment_cycle_path(@provider.provider_code, @provider.recruitment_cycle_year)
+    provider_breadcrumb << ["Request PE courses for 2021/22", path]
```
Comment (asmega) [inline app/helpers/breadcrumb_helper.rb:75]: is it possible to not hard code the year and work it out based on the current recruitment cycle year?

Comment (asmega) [reply app/helpers/breadcrumb_helper.rb:75]: or is that out of scope?

Comment (aliuk2012) [reply app/helpers/breadcrumb_helper.rb:75]: I was thinking the same thing but I think for the moment lets hard code it and refactor it later.

File: spec/features/providers/show_spec.rb [hunk 2]
```diff
@@ -18,6 +18,11 @@
       visit provider_path(provider.provider_code)
       expect { organisation_show_page.courses_as_accredited_body_link }.to raise_error(Capybara::ElementNotFound)
     end
+
+    it "does not have the request PE courses for 2021/22 link" do
+      visit provider_path(provider.provider_code)
+      expect { organisation_show_page.request_allocations_link }.to raise_error(Capybara::ElementNotFound)
```
Comment (defong) [inline spec/features/providers/show_spec.rb:24]: using exception as a verification mechanism is a bad test

https://www.rubydoc.info/gems/site_prism/frames#Testing_that_an_element_does_not_exist

File: spec/features/providers/show_spec.rb [hunk 3]
```diff
@@ -27,5 +32,23 @@
...
+      it "does have the request PE courses for 2021/22 link" do
+        visit provider_path(provider.provider_code)
+        expect(organisation_show_page.request_allocations_link.text).to eq("Request PE courses for 2021/22")
+      end
+    end
+
+    context "When the user is not an admin" do
+      it "does not have the request PE courses for 2021/22 link" do
+        visit provider_path(provider.provider_code)
+        expect { organisation_show_page.request_allocations_link }.to raise_error(Capybara::ElementNotFound)
```
Comment (defong) [inline spec/features/providers/show_spec.rb:50]: using exception as a verification mechanism is a bad test

File: spec/features/providers/show_spec.rb (hunk 2 above)
Comment (AgaDufrat) [reply spec/features/providers/show_spec.rb:24]: I agree, was just following the pattern in this file. I will change them all.

File: app/helpers/breadcrumb_helper.rb (hunk 1 above)
Comment (AgaDufrat) [reply app/helpers/breadcrumb_helper.rb:75]: I will add a card for this

File: config/routes.rb [hunk 4]
```diff
@@ -181,6 +181,8 @@
       end
 
       resources :sites, path: "locations", on: :member, except: %i[destroy show]
+
+      get "/allocations", on: :member, to: "providers/allocations#index"
```
Comment (defong) [inline config/routes.rb:185]:
```suggestion
      scope module: "providers" do
        resources :allocations, only: [:index]
      end
```

and all the crazy thing it enlists...

**Pull request merged**